### PR TITLE
refactor(checkbox): styles

### PR DESCRIPTION
**Pull request by slhay28**
## Description

Refactoring checkbox styles to reflect new styles configuration with toneModes

### CODE REVIEW

File: src/packages/solid/components/Checkbox/Checkbox.tsx [hunk 1]
```diff
@@ -30,10 +30,18 @@ export interface CheckboxProps extends NodeProps {
 const Checkbox: Component<CheckboxProps> = (props: CheckboxProps) => {
   return (
     <View
+      forwardStates
       {...props}
+      tone={props.tone ?? styles.tone}
+      style={props?.style?.Container ?? styles.Container}
```
Comment (chiefcll) [inline src/packages/solid/components/Checkbox/Checkbox.tsx:36]: Here we can start doing this now:
={[props.style?.Container, styles.Container]}

File: src/packages/solid/components/Checkbox/Checkbox.tsx [hunk 2]
```diff
@@ -30,10 +30,18 @@ export interface CheckboxProps extends NodeProps {
...
+      {...{
+        ...styles.Container[props.tone || styles.tone],
+        ...props?.style?.Container[props.tone || styles.tone]
```
Comment (chiefcll) [inline src/packages/solid/components/Checkbox/Checkbox.tsx:39]: props is always an object so no props? ever

File: src/packages/solid/components/Checkbox/Checkbox.tsx [hunk 3]
```diff
@@ -30,10 +30,18 @@ export interface CheckboxProps extends NodeProps {
...
+      }}
+      states={{
+        checked: props.checked ?? false
```
Comment (chiefcll) [inline src/packages/solid/components/Checkbox/Checkbox.tsx:42]: checked: props.checked is fine - undefined is falsy so you get same result

File: src/packages/solid/components/Icon/Icon.tsx [hunk 4]
```diff
@@ -49,9 +49,13 @@ const Icon: Component<IconProps> = props => {
...
     <View
       {...props}
-      style={props?.style?.Container ?? styles.Container}
       tone={props.tone ?? styles.tone}
-      states={props.tone ?? styles.tone}
+      style={props?.style?.Container ?? styles.Container}
+      {...{
+        ...styles.Container[props.tone || styles.tone],
+        ...props?.style?.Container[props.tone || styles.tone]
+      }}
```
Comment (chiefcll) [inline src/packages/solid/components/Icon/Icon.tsx:57]: @anthony9187 ^ - I'm just saying this is messy. With the new style change you can pass these into the array of styles...

Comment (anthony9187) [reply src/packages/solid/components/Icon/Icon.tsx:57]: totally agree - i hated this setup, and think this super-destructor should be buried and never spoken of again

File: src/packages/solid/components/Checkbox/Checkbox.tsx (hunk 1 above)
Comment (anthony9187) [reply src/packages/solid/components/Checkbox/Checkbox.tsx:36]: we should be able to include the below tone setup in that array as well, like we were saying in Icon

**Review by chiefcll — APPROVED** [review]

**Pull request merged**